app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -226,13 +236,34 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, resultData: Intent?) {
-        if (requestCode == REQUEST_CODE_FOR_PERFORM_SAF && resultCode == Activity.RESULT_OK) {
-            if (resultData != null) {
-                uploadFile(resultData.data)
+        if (resultData != null && resultCode == Activity.RESULT_OK) {
+            when(requestCode){
+                REQUEST_CODE_FOR_PERFORM_SAF -> {
+                    uploadFile(resultData.data)
+                }
+                REQUEST_CODE_FOR_DRAW -> {
+                    val result= resultData.getByteArrayExtra("bitmap")
+                    val bitmap = BitmapFactory.decodeByteArray(result, 0, result.size)
+                    val uri = saveImage(bitmap)
```

Comment:
Do you really need to salve the draw image every time the user uses this feature on device?

Reply:
@filipedelimabrito I'm saving the image so that I can get the Image Uri. Is there a way to get the bitmap uri without saving it?